### ISSUE TITLE
Compiling exercises, rewriting links converts quote escaping backslashes to common slashes

### ISSUE THREAD

**Issue by LaaksonenV** [Aalto-LeTech/a-plus#396]
After compiling all files, my exercise yaml in _build/yaml folder has a line like 
`instructions: "<p>\n  <iframe src=\"_static/ggframe.html\">`. Then, as compile reaches the mentioned phase of rewriting links, the line is changed to  
`instructions: "<p>\n  <iframe src=\"static/content/host/_static/ggframe.html/">`. The quote at the end of the link is no longer escaped, and poop meet fan.

**Comment by Mankro** [comment]
So, at the end there is a syntax error in the YAML string due to incorrect escaping of the quotes. This is actually an issue in the a-plus-rst-tools, not a-plus. The code for rewriting those links in the HTML and YAML files is very simplistic and does not handle escaping quotes in strings. This problem does not occur all the time since often the YAML strings use single quotes as delimiters instead of double quotes. Of course, the escaping issue should be fixed properly, but it has not been given much priority. 

Are you using the latest version a-plus-rst-tools? The code has never handled escaping, but I don't know why the YAML string is sometimes written in single quotes and sometimes in double quotes. Changing the version might affect that.

The problematic code is here: https://github.com/Aalto-LeTech/a-plus-rst-tools/blob/538c8da81a3eb62578999fd068cdf639e13614bc/lib/html_tools.py

**Comment by Mankro** [comment]
Another thing that you can change in your course: either set or do not set the STATIC_CONTENT_HOST variable. It affects how the link is manipulated in the end and you might avoid the escaping issue.

**Comment by LaaksonenV** [comment]
Thanks for a quick response, quicker than I was able to return to the problem.
I had previously encountered a similiar problem when a link was cut to a new line, and that was somehow corrected by rearranging the tag so the link wasn't cut, go figure...
Updating rst-tools didn't help.

STATIC_CONTENT_HOST was set by default, and while removing it did stop link rewriting, it introduced a new attribute before the src, 'data-aplus-path="/static/{course}" ', without escaped quotes, so that wasn't really a solution.

For now, I bodged the problem by editing line 102 in the html_tools.py to ingore all string delimiter escaping backslashes:
`                    out += content[i:j] + static_host + re.sub(r'\\[^\"\']','/',my_path) #previously ... + my_path.replace('\\','/')
`
